warehouse/core/src/main/java/datawave/ingest/mapreduce/job/RFileSplit.java

```diff
@@ -0,0 +1,75 @@
+package datawave.ingest.mapreduce.job;
+
+import java.io.DataInput;
+import java.io.DataOutput;
+import java.io.IOException;
+
+import org.apache.accumulo.core.data.Key;
+import org.apache.accumulo.core.data.Range;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.mapreduce.lib.input.FileSplit;
+
+// For a file with N blocks
+// startBlock 0 - beginning of the file to the first block
+// startBlock N - last block to the end
+public class RFileSplit extends FileSplit {
+    private long startBlock;
+    private long numBlocks;
+    private Key top;
+
+    public RFileSplit() {
+        top = new Key();
+    }
+
+    public RFileSplit(Path path, long fileStart, long fileLength, String[] hosts, int startBlock, int numBlocks, Key top) {
+        super(path, fileStart, fileLength, hosts);
+        this.startBlock = startBlock;
```

Comment:
We're converting `startBlock` and `numBlocks` from an int to a long. Do we need it to be a long?

Reply:
these should all be longs, its very unlikely, but possible to extend this many blocks